hub/api/info.py

```diff
@@ -132,6 +84,53 @@ def __getattr__(self, key: str):
                 return self._info
             return self[key]
 
+    def __setattr__(self, key: str, value):
+        if key in {"_info", "_dataset", "_key", "is_dirty"}:
+            object.__setattr__(self, key, value)
+        else:
+            with self:
+                self[key] = value
+
+    def get(self, key, default=None):
+        return self._info.get(key, default)
+
+    def setdefault(self, key, default=None):
+        with self:
+            ret = self._info.setdefault(key, default)
+        return ret
+
+    def clear(self):
+        with self:
+            self._info.clear()
+
+    def pop(self, key, default=None):
+        with self:
+            popped = self._info.pop(key, default)
+        return popped
+
+    def popitem(self):
+        with self:
+            popped = self._info.popitem()
```

Comment:
In python < 3.7, `popitem()` pops and returns an arbitrary entry while in 3.7+ it returns LIFO. Have you taken that into account wherever `popitem` is called?  (Not just info, but lru_sizes.popitem etc as well)

Reply:
For info, it shouldn't matter, it will behave like the corresponding version's dictionary.
For LRUCache, I don't think it should be an issue even in 3.6 as the Cpython implementation retains order.

From StackOverflow (https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6):-
"They are insertion ordered. As of Python 3.6, for the CPython implementation of Python, dictionaries remember the order of items inserted. This is considered an implementation detail in Python 3.6; you need to use OrderedDict if you want insertion ordering that's guaranteed across other implementations of Python (and other ordered behavior).

As of Python 3.7, this is no longer an implementation detail and instead becomes a language feature."